prime-router/src/main/resources/metadata/hl7_mapping/resources/Observation/OBX.yml

```diff
@@ -59,6 +59,7 @@ elements:
 
   - name: obx-equipment-instance-identifier
     resource: '%resource.device.resolve().identifier.union(%resource.extension(%`rsext-obx-observation`).extension.where(url = "OBX.18").tail().value.resolve().identifier)'
+    condition: '%resource.value.exists() or %resource.extension(%`rsext-assigning-authority`).exists()'
```

Comment:
This condition handles us getting sent FHIR like so 
```
{
    "fullUrl" : "Device/1710881998384474000.9752cf60-d1b6-4cbe-98b2-9984718ee4ed",
    "resource" : {
      "resourceType" : "Device",
      "id" : "1710881998384474000.9752cf60-d1b6-4cbe-98b2-9984718ee4ed",
      "identifier" : [ {
        "extension" : [ {
          "url" : "https://reportstream.cdc.gov/fhir/StructureDefinition/assigning-authority",
          "extension" : [ {
            "url" : "https://reportstream.cdc.gov/fhir/StructureDefinition/namespace-id",
            "valueString" : "Equipment"
          }, {
            "url" : "https://reportstream.cdc.gov/fhir/StructureDefinition/universal-id",
            "valueString" : "1.111.122311.222.44.2.3.3"
          }, {
            "url" : "https://reportstream.cdc.gov/fhir/StructureDefinition/universal-id-type",
            "valueCode" : "ISO"
          } ]
        } ],
        "value" : "CDC_3015822725"
      } ]
    }
  }
```
where the second identifier is missing a value

Reply:
I'm only seeing a one identifier in this snippet. Where's the second identifier? Also the resource on the element references `extension.where(url = "OBX.18")` but I don't see that extension in the test FHIR data. Is that where the second identifier should come from?

Reply:
so `%resource.device.resolve().identifier` will grab all the identifiers on the device and then map them, so in this case there are no values in the OBX.18 extension.

That case actually only comes up when mapping from HL7->FHIR->HL7 since `device` has cardinality of 1